test/examples/star-cors-example.test.ts

```diff
@@ -29,4 +29,23 @@ describe('Star CORS examples', () => {
 
     response.headers['access-control-allow-origin'].should.eql('http://localhost:3000');
   });
+
+  it('/api triggers cors policy', async () => {
+    const response = await supertest('localhost:3000')
+      .get('/api')
+      .set('origin', 'http://localhost.foreign.com')
+      .expect(200);
+
+    response.headers['access-control-allow-origin'].should.not.eql('http://localhost.foreign.com');
+    response.headers['access-control-allow-origin'].should.eql('localhost');
+  });
+
+  it('/api/example returns access-control-allow-origin that contains the subdomain', async () => {
+    const response = await supertest('localhost:3000')
+      .get('/api/example')
+      .set('origin', 'http://some.deep.subdomain.localhost')
+      .expect(200);
+
+    response.headers['access-control-allow-origin'].should.eql('http://some.deep.subdomain.localhost');
```

Comment:
❓ Why does this work? It shouldn't work right? Would it work with the previous implementation?